Sources/PayPalNativePayments/NativeCheckoutProvider.swift

```diff
@@ -7,27 +7,74 @@ import CorePayments
 
+    private let checkout: CheckoutProtocol.Type
+    
+    init(_ mxo: CheckoutProtocol.Type = Checkout.self) {
+        self.checkout = mxo
+    }
     // swiftlint:disable:next function_parameter_count
     func start(
-        presentingViewController: UIViewController? = nil,
-        createOrder: CheckoutConfig.CreateOrderCallback?,
-        onApprove: CheckoutConfig.ApprovalCallback?,
-        onShippingChange: CheckoutConfig.ShippingChangeCallback?,
-        onCancel: CheckoutConfig.CancelCallback?,
-        onError: CheckoutConfig.ErrorCallback?,
+        presentingViewController: UIViewController?,
+        orderID: String,
+        onStartableApprove: @escaping StartabeApproveCallback,
+        onStartableShippingChange: @escaping StartabeShippingCallback,
+        onStartableCancel: @escaping StartabeCancelCallback,
+        onStartableError: @escaping StartabeErrorCallback,
         nxoConfig: CheckoutConfig
     ) {
-        Checkout.showsExitAlert = false
-        Checkout.set(config: nxoConfig)
-        DispatchQueue.main.async {
-            Checkout.start(
+        checkout.showsExitAlert = false
+        checkout.set(config: nxoConfig)
+        guaranteeMainThread {
+            self.checkout.start(
                 presentingViewController: presentingViewController,
-                createOrder: createOrder,
-                onApprove: onApprove,
-                onShippingChange: onShippingChange,
-                onCancel: onCancel,
-                onError: onError
+                createOrder: { createOrderAction in
+                    createOrderAction.set(orderId: orderID)
+                },
+                onApprove: { approval in
+                    onStartableApprove(approval.data.ecToken, approval.data.payerID)
+                },
+                onShippingChange: { shippingChangeData, shippingChangeActions in
+                    let type = shippingChangeData.type
+                    let shippingActions = PayPalNativePaysheetActions(shippingChangeActions)
+                    let shippingAddress = PayPalNativeShippingAddress(shippingChangeData.selectedShippingAddress)
+                    var shippingMethod: PayPalNativeShippingMethod?
+                    if let selectedMethod = shippingChangeData.selectedShippingMethod {
+                        shippingMethod = PayPalNativeShippingMethod(selectedMethod)
+                    }
+                    onStartableShippingChange(type, shippingActions, shippingAddress, shippingMethod)
+                },
+                onCancel: { onStartableCancel() },
+                onError: { error in
+                    onStartableError(error.reason)
+                }
             )
         }
     }
+    
+    private func guaranteeMainThread(_ work: @escaping () -> Void) {
+        if Thread.isMainThread {
+            work()
+        } else {
+            DispatchQueue.main.async(execute: work)
+        }
+    }
 }
+
+protocol CheckoutProtocol {
```

Comment:
I think we can remove `NativeCheckoutStartable` now that we have `CheckoutProtocol` which we use to access `start()` via `checkout.start()` in `NativeCheckoutProvider`